tests/main/views/test_direct_award.py

```diff
@@ -659,29 +666,98 @@ def _lock_project_data(self):
+        with self.app.app_context():
+            project = DirectAwardProject.query.get(self.project_id)
+            project.locked_at = datetime.utcnow()
+
+            db.session.add(project)
+            db.session.commit()
+
+        res = self.client.post(
+            '/direct-award/projects/{}/lock'.format(self.project_id),
+            data=json.dumps({
+                'updated_by': 'example',
+            }),
+            content_type='application/json')
+
+        assert res.status_code == 400
 
-    @pytest.mark.skip
     def test_lock_project_404s_if_invalid_project(self):
-        pass
+        res = self.client.post(
+            '/direct-award/projects/{}/lock'.format("not_a_valid_id"),
+            data=json.dumps({
+                'updated_by': 'example',
+            }),
+            content_type='application/json')
 
-    @pytest.mark.skip
-    def test_lock_project_search_and_project_datetimes_are_updated(self):
-        pass
+        assert res.status_code == 404
 
-    @pytest.mark.skip
-    def test_lock_project_search_result_entries_populated_with_latest_archived_service_for_searched_services(self):
-        pass
+    @mock.patch('app.main.views.direct_award.search_api_client')
+    def test_lock_project_success(self, search_api_client):
+        self._create_service()
+        service_id = "1234567890123458"
+        search_api_client.search_services_from_url_iter.return_value = [{"id": service_id}]
+
+        res = self.client.post(
+            '/direct-award/projects/{}/lock'.format(self.project_id),
+            data=json.dumps({
+                'updated_by': 'example',
+            }),
+            content_type='application/json')
+        data = json.loads(res.get_data(as_text=True))
 
-    @pytest.mark.skip
-    def test_lock_project_saved_result_entries_are_for_the_active_search_in_the_project(self):
-        pass
+        assert res.status_code == 200
+        assert data['project']['id'] == self.project_id
+        assert data['project']['lockedAt'] is not None
 
-    @pytest.mark.skip
-    def test_lock_project_creates_audit_event(self):
-        pass
+        with self.app.app_context():
+            assert AuditEvent.query.order_by(AuditEvent.id.desc()).first().type == AuditTypes.lock_project.value
+            search_result_entry = DirectAwardSearchResultEntry.query.filter(
+                DirectAwardSearchResultEntry.search_id == self.search_id
+            )
+            archived_services = ArchivedService.query.\
+                filter(ArchivedService.service_id.in_([service_id])).\
+                group_by(ArchivedService.id).\
+                order_by(ArchivedService.service_id, desc(ArchivedService.id)).\
+                distinct(ArchivedService.service_id).all()
+
+            assert len(archived_services) > 0
+            assert search_result_entry.count() == 1
+            assert search_result_entry.all()[0].archived_service_id == archived_services[0].id
+
+    def _create_service(self):
```

Comment:
This should be moved higher up (i.e probably after setup). Also it's creating and updating a service, so probably call it `_create_service_and_update`.